cmd/modelcmd/apicontext.go

```diff
@@ -17,11 +19,19 @@ import (
 	"github.com/juju/juju/jujuclient"
 )
 
+type CookieJar interface {
+	http.CookieJar
+	RemoveAllHost(host string)
+}
+
 // APIContext holds the context required for making connections to
 // APIs used by juju.
 type APIContext struct {
-	Jar            *cookiejar.Jar
-	WebPageVisitor httpbakery.Visitor
+	Jar CookieJar
+	// jar holds the internal version of the cookie jar - it has
+	// methods that clients should not use, such as Save.
+	jar            *domainCookieJar
```

Comment:
having a member "Jar" and another member "jar" sounds like a *huge* chance to make mistakes in using the right one.
Can we clarify these more with something like 'domainJar' at least?
I'm not sure why one is exported and one isn't, and comments like
"we send it to everyone because we don't know who needs it", feels very much like a security hole.